mapper/container_ports.go

```diff
@@ -98,13 +98,11 @@ func (n ContainerPorts) ToInternal(data map[string]interface{}) error {
 					logrus.Warnf("Failed to encode port: %v", err)
 					return obj
 				}
-				if strings.EqualFold(convert.ToString(mapped["kind"]), "HostPort") {
-					if _, ok := mapped["sourcePort"]; ok {
-						mapped["hostPort"] = mapped["sourcePort"]
-					}
+				if !strings.EqualFold(convert.ToString(mapped["kind"]), "HostPort") {
+					// delete the source port so it doesn't get converted to the host port by default mapper
+					delete(mapped, "sourcePort")
 				}
-				// delete the source port so it doesn't get converted to the host port by default mapper
-				delete(mapped, "sourcePort")
+
```

Comment:
should we also have an integration test to check this usecase of non-native hostport is preserved?

Reply:
good idea @prachidamle

Reply:
@prachidamle added the test https://github.com/rancher/rancher/pull/23974